apps/system/js/core.js

```diff
@@ -50,6 +54,85 @@
     },
 
     _start: function() {
+      ScreenManager && ScreenManager.turnScreenOn();
+      // We need to be sure to get the focus in order to wake up the screen
+      // if the phone goes to sleep before any user interaction.
+      // Apparently it works because no other window
+      // has the focus at this point.
+      window.focus();
+      // With all important event handlers in place, we can now notify
+      // Gecko that we're ready for certain system services to send us
+      // messages (e.g. the radio).
+      // Note that shell.js starts listen for the mozContentEvent event at
+      // mozbrowserloadstart, which sometimes does not happen till
+      // window.onload.
+      this.publish('mozContentEvent', {
+        type: 'system-message-listener-ready'
+      }, true);
+
+      this.loadWhenIdle([
+        'CameraTrigger',
+        'NotificationScreen',
+        'AirplaneMode',
+        'NotificationsSystemMessage',
+        'Accessibility',
+        'AlarmMonitor',
+        'DebuggingMonitor',
+        'TimeCore',
+        'GeolocationCore',
+        'TetheringMonitor',
+        'UsbCore',
+        'TextSelectionDialog',
+        'ExternalStorageMonitor',
+        'DeviceStorageWatcher',
+        'AppUsageMetrics',
+        'CellBroadcastSystem',
+        // This should be loaded by MobileConnectionCore.
+        // However, the integration test is testing this on desktop b2g
+        // which has no navigator.mozMobileConnections.
+        'CpuManager',
+        'HomeGesture',
+        'SourceView',
+        'TtlView',
+        'MediaRecording',
+        'QuickSettings',
+        'UsbStorage',
+        'MobileIdManager',
+        'FindmydeviceLauncher',
+        'FxAccountsManager',
+        'FxAccountsUI',
+        'NetworkActivity',
+        'CrashReporter',
+        'Screenshot',
+        'SoundManager',
+        'CustomDialogService',
+        'CarrierInfoNotifier'
+        // XXX: We should move this into mobileConnectionCore,
+        // but integration tests running on desktop without mobileConnection
+        // is testing this.
+      ]).then(function() {
+        this.startAPIHandlers();
+        return Promise.resolve();
```

Comment:
nit: somebody will correct me but I don't think you _have_ to return a promise here.